src/evm/stream.rs

```diff
@@ -21,6 +21,19 @@ pub struct ProtocolStreamBuilder {
     stream_builder: TychoStreamBuilder,
 }
 
+/// Builds the protocol stream, providing a `BlockUpdate` for each block received.
+///
+/// Each `BlockUpdate` can then be used at a higher level to retrieve important information from
+/// the block, such as the updated states of protocol components, which can in turn be used
+/// to obtain spot price information for a desired component and token pair.
+///
+/// # Returns
+/// A result containing a stream of decoded block updates, where each item is either:
+/// - `Ok(BlockUpdate)` if decoding succeeds.
+/// - `Err(StreamDecodeError)` if a decoding error occurs.
+///
+/// # Errors
+/// Returns a `StreamError` if the underlying stream builder fails to initialize.
```

Comment:
Please add a section here like `# Important` that briefly touches on how decoding works (e.g. if tokens are missing, the pool is skipped, etc.) You can do that by linking to the decoder struct docs.